test/api_api_tokens_test.go

```diff
@@ -0,0 +1,141 @@
+
+		expireDate, err := time.Parse(time.RFC3339, "2021-01-01T00:00:00.000Z")
+		require.Nil(t, err)
+		token := client.CreateApiTokenSchemaOneOf{
+			Type:      "admin",
+			TokenName: "test admin token",
+			ExpiresAt: &expireDate,
+		}
+		createTokenSchema := client.CreateApiTokenSchemaOneOfAsCreateApiTokenSchema(&token)
+		resp, httpRes, err := apiClient.APITokensAPI.CreateApiToken(context.Background()).CreateApiTokenSchema(createTokenSchema).Execute()
+
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		assert.Equal(t, 201, httpRes.StatusCode)
+
+		createdTokens = append(createdTokens, resp.Secret)
+	})
+
+	t.Run("Test APITokensAPIService CreateApiToken (client)", func(t *testing.T) {
+
+		expireDate, err := time.Parse(time.RFC3339, "2021-01-01T00:00:00.000Z")
+		require.Nil(t, err)
+		environment := "development"
+		token := client.CreateApiTokenSchemaOneOf2{
+			Type:        "client",
+			TokenName:   "test client token",
+			ExpiresAt:   &expireDate,
+			Environment: &environment,
+		}
+		createTokenSchema := client.CreateApiTokenSchemaOneOf2AsCreateApiTokenSchema(&token)
+		resp, httpRes, err := apiClient.APITokensAPI.CreateApiToken(context.Background()).CreateApiTokenSchema(createTokenSchema).Execute()
+
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		assert.Equal(t, 201, httpRes.StatusCode)
+
+		createdTokens = append(createdTokens, resp.Secret)
+	})
+
+	t.Run("Test APITokensAPIService CreateApiToken (frontend)", func(t *testing.T) {
+
+		expireDate, err := time.Parse(time.RFC3339, "2021-01-01T00:00:00.000Z")
+		require.Nil(t, err)
+		environment := "development"
+		project := "default"
+		token := client.CreateApiTokenSchemaOneOf2{
+			Type:        "frontend",
+			TokenName:   "test frontend token",
+			ExpiresAt:   &expireDate,
+			Environment: &environment,
+			Project:     &project,
+		}
+		createTokenSchema := client.CreateApiTokenSchemaOneOf2AsCreateApiTokenSchema(&token)
+		resp, httpRes, err := apiClient.APITokensAPI.CreateApiToken(context.Background()).CreateApiTokenSchema(createTokenSchema).Execute()
+
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		assert.Equal(t, 201, httpRes.StatusCode)
+
+		createdTokens = append(createdTokens, resp.Secret)
+	})
+
+	t.Run("Test APITokensAPIService DeleteApiToken", func(t *testing.T) {
+
+		for i := 0; i < len(createdTokens); i++ {
+			token := createdTokens[i]
+
+			httpRes, err := apiClient.APITokensAPI.DeleteApiToken(context.Background(), token).Execute()
+
+			require.Nil(t, err)
+			assert.Equal(t, 200, httpRes.StatusCode)
+		}
+
+	})
+
+	t.Run("Test APITokensAPIService GetAllApiTokens", func(t *testing.T) {
+
+		resp, httpRes, err := apiClient.APITokensAPI.GetAllApiTokens(context.Background()).Execute()
+
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		assert.Equal(t, 200, httpRes.StatusCode)
+
+	})
+
+	t.Run("Test APITokensAPIService GetApiTokensByName", func(t *testing.T) {
+
+		t.Skip("skip test") // remove to run test
```

Comment:
It doesn't work, I even tried this with curl... I was surprised but didn't want to go into debugging mode... My first idea was that this wouldn't be useful, but it might be for the `data` definition :thinking:

Reply:
No no, makes sense, was just checking if the code under test was borked or if this didn't get spotted!

Reply:
Added [e599b84](https://github.com/Unleash/unleash-server-api-go/pull/12/commits/e599b84279bf4d6d0ddd0b2b120d4fa925b81bad). Turns out it's a new API and I hadn't updated my docker image of unleash.